### ISSUE TITLE
Different Configuration modes and Val loss

### ISSUE THREAD

**Issue by mkhoshle** [Arthur151/ROMP#250]
Hi, 

Q1. In the code, I can see that there are different configuration modes: (`matching_gts`,`parsing`,`forward`). Wanted to ask what these modes should be during train, validation, and test?  Based on the paper seems like `matching_gts` is used for training however it is not clear what modes are used during validation and test?

Q2. Also in `base.py` you have:
```
self.eval_cfg = {'mode':'matching_gts', 'is_training':False, 'calc_loss': False}
self.val_cfg = {'mode':'parsing', 'calc_loss': True}
```
What is the difference between `self.eval_cfg` and `self.val_cfg`? Which one is for test and which one is for validation?

Q3. Also regarding the loss function, looking at your logs [here](https://github.com/Arthur151/ROMP/files/7742306/V1_hrnet_nopretrain_h36m.mpiinf.coco.mpii.lsp.muco.crowdpose_g0.1.2.3.log), seems like only training loss is logged. Do you calculate validation loss as well? I can see that some metrics are calculated during validation. However, what do I need to do if I want to get validation loss?

Q4. Also, why is the training loss value large? I was training the code and getting similar loss values but I was thinking that I am doing something wrong until I saw your logs in the Github.

Thanks,

**Comment by Arthur151** [comment]
Hi, Mahzad Khoshlessan,

About Q1 & Q2.
During evaluation, our goal is to evaluate the 3D mesh / pose error of predictions. 
To achieve this, we need to match the preditions with the ground truth via `matching_gts`. 
Via matching the predictions with the ground truth, we are able to supervised the predictions of each person. 
Via matching the predictions with the ground truth, we are able to calculate the matrix evalution results. 
Therefore, we use the `matching_gts` for training & most evaluation.
`self.val_cfg` is not the typical validation as usual. It just how we set to get the pure inference results. We don't need to match with gts if we just want to get the results. 

Q3. I guess the validation loss you mean is to see how the trained model performs on validation set. Normally, we don't train on validation set, therefore, we don't calculate the loss on validation set. To see how the trained model performs on validation set, we evaluate the model on it. Current code is doing this every 2000 iterations. 

Q4. The loss value is mutiplied with a weight value to make it significant enough for effective learning. The large weights at 
https://github.com/Arthur151/ROMP/blob/95bdf66634f1ab0c9f38aad97d18f71c8008291d/configs/v1.yml#L40
make them large.

**Comment by Arthur151** [comment]
Hi, @mkhoshle 

About overfitting problem, we can compare the performance on validation/test set. The loss would be always smaller on training set than that of validation set. As long as the performance on validation/test set is great then I think we are fine. 

About the model verison, I just release the V1 of ROMP. V6 stands for training code of BEV, which has been released. I just release the inference code of BEV in simple_romp. I will release the training code of BEV within a month.
